Added dependency of crowdsec to traefik and relaxed its health check

TuncTaylan opened the pull request:
## Community Contribution License Agreement
By creating this pull request, I grant the project maintainers an unlimited,
perpetual license to use, modify, and redistribute these contributions under any terms they
choose, including both the AGPLv3 and the Fossorial Commercial license terms. I
represent that I have the right to grant this license for all contributed content.

## Description
Add dependancy on crowdsec for traefik and more relaxed health check for crowdsec, to prevent zombie traefik bouncers when starting the stack up.

Otherwise there are always "loose" or duplicate traefik bouncer plugins (check the last api pull dates):

```bash
docker exec -it crowdsec cscli bouncers list
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 Name                        IP Address  Valid  Last API pull         Type                             Version                                                                  Auth Type
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 traefik-bouncer             172.18.0.4  ✔️     2025-04-21T09:24:34Z  Crowdsec-Bouncer-Traefik-Plugin  1.X.X                                                                    api-key
 vps-firewall                172.18.0.1  ✔️     2025-04-21T09:32:16Z  crowdsec-firewall-bouncer        v0.0.31-debian-pragmatic-amd64-4b99c161b2c1837d76c5fa89e1df83803dfbcc87  api-key
 traefik-bouncer@172.18.0.3  172.18.0.3  ✔️     2025-04-18T20:43:20Z  Crowdsec-Bouncer-Traefik-Plugin  1.X.X                                                                    api-key
 traefik-bouncer@172.18.0.2  172.18.0.2  ✔️     2025-04-15T06:55:59Z  Crowdsec-Bouncer-Traefik-Plugin  1.X.X                                                                    api-key
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```


## How to test?
Just run the installer and do a docker-compose.yaml comparison.

oschwartz10612 commented on install/config/docker-compose.yml:
```diff
@@ -48,6 +48,10 @@ services:
+{{if .DoCrowdsecInstall}}
+      crowdsec:
+        condition: service_healthy
+{{end}}
```
This actually wont work because of the order of operations of the install and if you are doing an install after the fact once this .yml is already created.

TuncTaylan replied:
Yeah, sorry. I didn't invest much time to make sense, how the install routine worked.

oschwartz10612 commented:
Thanks for the contribution!

We need to think a little bit about how to get the condition into the main .yml file. I would do it in a similar way I am updating the log volume in `CheckAndAddTraefikLogVolume` in the `config.go` file. I think we would need a function to check if the condition is there or not and only add it if its is needed and then call this function in the setup crowdsec workflow. Does that make sense?

TuncTaylan commented:
> Thanks for the contribution!
> 
> We need to think a little bit about how to get the condition into the main .yml file. I would do it in a similar way I am updating the log volume in `CheckAndAddTraefikLogVolume` in the `config.go` file. I think we would need a function to check if the condition is there or not and only add it if its is needed and then call this function in the setup crowdsec workflow. Does that make sense?

Yes, it makes absolutely sense. Is there any pseudo code or list of steps about how your installation routine works or what steps it needs to go through, in order to install or update?

I'm pretty sure, that it would be possible to do with Helm charts or Ansible scripts.

TuncTaylan commented:
> Thanks for the contribution!
> 
> We need to think a little bit about how to get the condition into the main .yml file. I would do it in a similar way I am updating the log volume in `CheckAndAddTraefikLogVolume` in the `config.go` file. I think we would need a function to check if the condition is there or not and only add it if its is needed and then call this function in the setup crowdsec workflow. Does that make sense?

Yes, that make sense. I'm now confident enough, that it works as intended.
I've refactored some of the existing recurring bits and added the dependency with a function "CheckAndAddCrowdsecDependency"
